src/App.js

```diff
@@ -34,6 +41,28 @@ class App extends Component {
     this.setState({
       initialTransactions: transactions
     });
+    const uniquePartiesList = this.filterTransactionList(transactions);
+    this.updateUniquePartiesList(uniquePartiesList);
+  };
+
+  /**
+   * @desc filterTransactionList removes duplicate transactions from transactions list.
+   * @param initialTransactionsList
+   * @return list of unique transactions based on the party
+   */
+  filterTransactionList = initialTransactionsList => {
```

Comment:
`App.js` is getting quite cluttered so `filterTransactionList()` could be moved to `utils.js`